### PR TITLE
feat(component): SfTile

### LINKED ISSUE #1127: [FEATURE] (molecule) SfTile
**Describe how this feature will help us improve**
Describe why do you think this feature is needed. 
Zrzut ekranu 2020-04-28 o 16 17 17
Zrzut ekranu 2020-04-28 o 16 17 36

**Describe the solution you'd considered**
API 
``` vue
<SfTile
:title=""
:background="for colo"
:image="for image"
/>
```
should be similar to SfBanner/ SfCallToAction

### PR DESCRIPTION
**Pull request by justyna-13**
# Related issue
Closes #1127

# Scope of work
New component - SfTile

# Screenshots of visual changes

Zrzut ekranu z 2020-05-04 12-27-51
Zrzut ekranu z 2020-05-04 12-27-59

### CODE REVIEW

**Comment by martaradziszewska** [comment]
@justyna-13 @pspaczek Following the idea of making reusable components what do you think to make this component some kind of a base tile with default slot only with the possibility to add a border, shadow or some basic styles and modifiers like `--full-width`. You could build with this anything that looks like a tile by putting inside buttons, select, dropdowns, images - In fact our other components.  It can be SfTile with a lot of ways to modify it.

File: packages/vue/src/components/molecules/SfTile/SfTile.vue [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    <slot name="title" v-bind="{ title }">
+      <h3 v-if="title" class="sf-tile__title">
+        {{ title }}
+      </h3>
+    </slot>
+  </div>
+</template>
+<script>
+export default {
+  name: "SfAlert",
```
Comment (filrak) [inline packages/vue/src/components/molecules/SfTile/SfTile.vue:13]:
```suggestion
  name: "SfTile",
```

File: packages/vue/src/components/molecules/SfTile/SfTile.vue [hunk 2]
```diff
@@ -0,0 +1,55 @@
+<template>
+  <div class="sf-tile" :style="style">
+    <!-- @slot Use this slot to replace title -->
+    <slot name="title" v-bind="{ title }">
+      <h3 v-if="title" class="sf-tile__title">
```
Comment (filrak) [inline packages/vue/src/components/molecules/SfTile/SfTile.vue:5]: Why `<h3>` ? Shouldn't that be a div?

File: packages/vue/src/components/molecules/SfTile/SfTile.vue [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+     * Tile title
+     */
+    title: {
+      type: String,
+      default: "",
+    },
+    /**
+     * Tile background color
+     */
+    background: {
```
Comment (filrak) [inline packages/vue/src/components/molecules/SfTile/SfTile.vue:25]: Imho that should be CSS. You're probably refering to Banner component but in that case setting background is as easy as setting CSS property

```css
.tile-class {
  background: pink;
```

File: packages/vue/src/components/molecules/SfTile/SfTile.vue [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      type: String,
+      default: "",
+    },
+    /**
+     * Tile image
+     */
+    image: {
```
Comment (filrak) [inline packages/vue/src/components/molecules/SfTile/SfTile.vue:32]: this should be called background ;)

File: packages/vue/src/components/molecules/SfTile/SfTile.vue [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      type: [String, Object],
+      default: "",
+    },
+  },
+  computed: {
+    style() {
```
Comment (filrak) [inline packages/vue/src/components/molecules/SfTile/SfTile.vue:38]: Can you explain why we need that?

Comment (justyna-13) [reply packages/vue/src/components/molecules/SfTile/SfTile.vue:36]: it's for readable logic for background img :)

File: packages/shared/styles/components/molecules/SfTile.scss [hunk 6]
```diff
@@ -0,0 +1,38 @@
+@import "../../helpers";
+.sf-tile {
+  position: relative;
+  width: var(--tile-width, 6.25rem);
+  height: var(--tile-height, 7.5rem);
+  margin: var(--tile-margin, 0);
+  background-image: var(--tile-background-image);
+  background-repeat:  var(--tile-background-repeat, no-repeat);
+  background-color: var(--tile-background-color, transparent);
```
Comment (AdamPawlinski) [inline packages/shared/styles/components/molecules/SfTile.scss:9]: We can use background mixin here.

File: packages/shared/styles/components/molecules/SfTile.scss [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    color: var(--tile-title-color, var(--c-black));
+    transition: color 150ms ease-in-out;
+    @include font(
+      --font-tile,
+      var(--font-bold),
+      var(--h3-font-size),
+      1.6,
+      var(--font-family-secondary)
+    )
+  }
```
Comment (AdamPawlinski) [inline packages/shared/styles/components/molecules/SfTile.scss:26]: We can adjust variables to new styleguide.

**Review by AdamPawlinski — COMMENTED** [review]
Great. I would add also information to migration guide.

File: packages/shared/styles/components/molecules/SfTile.scss (hunk 6 above)
Comment (AdamPawlinski) [reply packages/shared/styles/components/molecules/SfTile.scss:9]: added

File: packages/shared/styles/components/molecules/SfTile.scss (hunk 7 above)
Comment (AdamPawlinski) [reply packages/shared/styles/components/molecules/SfTile.scss:26]: adjusted

**Review by AdamPawlinski — DISMISSED** [review]

**Comment by filrak** [comment]
I'm putting this temporarily on hold - there are new designs coming ;)

**Comment by mayashavin** [comment]
@justyna-13 can you please resolve conflict and update story?

File: packages/vue/src/components/molecules/SfTile/SfTile.vue [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+  computed: {
+    ...mapMobileObserver(),
+    style() {
+      if (typeof this.background === String) {
+        return {
+          background: `url('${this.background}')`,
+        };
+      } else {
+        return {
+          background: this.isMobile
```
Comment (mayashavin) [inline packages/vue/src/components/molecules/SfTile/SfTile.vue:59]: background: `url('${this.background[ this.isMobile ? "mobile" : "desktop" ]}')`

**Review by mayashavin — APPROVED** [review]
See my comments

File: packages/vue/src/components/molecules/SfTile/SfTile.vue (hunk 8 above)
Comment (justyna-13) [reply packages/vue/src/components/molecules/SfTile/SfTile.vue]: done:)

**Review by mayashavin — APPROVED** [review]

**Pull request merged**